harvester/utils.py

```diff
@@ -149,3 +151,51 @@ def assign_bindings_to_images(bindings, max_bindings_per_image, shared_prefix=""
             assign_bindings_to_images(prefixed_bindings, max_bindings_per_image, prefix)
         )
     return images
+
+
+def image_for_binding(dc_identifier, prefixes):
+    """
+    Find the image of which prefix matches the prefix of the given binding ID.
+    """
+    binding_id = binding_id_from_dc(dc_identifier)
+    matches = [re.search(f"^{prefix}", binding_id) for prefix in prefixes]
+    image = [match.group(0) for match in matches if match][0]
```

Comment:
@aajarven I would also favour your solution here, it is easier to read and a bit faster.

Reply:
I think I would still add the error raising. It should of course never be needed, but if it would, and we don't have it, it's either going to produce some likely hard-to-debug KeyError or such later in the process which is not good, or what would be even worse would be that it just slips through the process and we just didn't download some binding(s).

We could define a descriptive custom error class, but just going with `ValueError` and a descriptive message would be good too.

Reply:
Sorry, I completely forgot about the raising error part. Fixed now (for real this time).